e107_plugins/chatbox_menu/chatbox_menu_shortcodes.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/*
+ * e107 website system
+ *
+ * Copyright (C) 2008-2013 e107 Inc (e107.org)
+ * Released under the terms and conditions of the
+ * GNU General Public License (http://www.gnu.org/licenses/gpl.txt)
+ *
+ * e107 chatbox_menu Plugin
+ *
+*/
+if ( ! defined('e107_INIT')) {
+	exit;
+}
+
+
+class chatbox_menu_shortcodes extends e_shortcode
+{
+	/**
+	 *  Initializer for chatbox_menu_shortcodes class
+	 */
+	public function init()
+	{
+		if ( ! $this->var['user_image'] ) {
+			$this->addVars($this->retrieveUserDataByNick($this->var['cb_nick']));
+		}
+
+	}
+
+
+	/**
+	 * Fetches user data from user table
+	 *
+	 * @param $nick
+	 *
+	 * @return array|string
+	 */
+	protected function retrieveUserDataByNick($nick)
+	{
+		$sql = e107::getDb();
+
+		$tmp = explode('.', $nick);
+		$userId = (int)$tmp[0];
+
+		return $sql->retrieve('user', 'user_id, user_name, user_image',
+			"user_id = {$userId}");
+
+	}
+
```

Comment:
This has no cache, so it could lead to a lot of queries. Use `return e107::user($userId);` instead.